datumaro/components/dataset.py

```diff
@@ -911,6 +911,9 @@ def detect(path: str, env: Optional[Environment] = None,
         if env is None:
             env = Environment()
 
+        if not osp.exists(path):
+            raise FileNotFoundError(path)
```

Comment:
Consider moving it to `Environment.detect_dataset()`

Reply:
I moved it, but it looks like it will affect the #576 a bit.